src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -56,7 +58,37 @@ public static ConfidentialClientApplicationBuilder Create(string clientId)
         /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
         public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 certificate)
         {
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            Config.ClientCredentialCertificate = certificate;
+            Config.ConfidentialClientCredentialCount++;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the certificate associated with the application along with the specific claims to sign.
+        /// </summary>
+        /// <param name="certificate">The X509 certificate used as credentials to prove the identity of the application to Azure AD.</param>
+        /// <param name="claimsToSign">The claims to be signed by the provided certificate.</param>
+        /// <remarks>You should use certificates with a private key size of at least 2048 bytes. Future versions of this library might reject certificates with smaller keys. </remarks>
+        public ConfidentialClientApplicationBuilder WithClaims(X509Certificate2 certificate, IDictionary<string, string> claimsToSign)
```

Comment:
I wonder if we would not want to name this method: `WithClientClaims`, as, contrary to the .WithClaims() method on the AcquireTokenXX methods (which can be about the user), the claims here are purely client claims.